src/inet/TCPEndPoint.cpp

```diff
@@ -380,67 +380,53 @@ INET_ERROR TCPEndPoint::Connect(IPAddress addr, uint16_t port, InterfaceId intf)
 
 #if WEAVE_SYSTEM_CONFIG_USE_SOCKETS
 
+    char intfStr[64];
+
+    res = GetSocket(addrType);
+    if (res != INET_NO_ERROR)
+        return res;
+
     if (intf == INET_NULL_INTERFACEID)
     {
         // The behavior when connecting to an IPv6 link-local address without specifying an outbound
         // interface is ambiguous. So prevent it in all cases.
         if (addr.IsIPv6LinkLocal())
             return INET_ERROR_WRONG_ADDRESS_TYPE;
     }
-
-#if INET_CONFIG_ENABLE_IPV4
     else
     {
-        // Attempting to initiate an IPv4 connection via a specific interface is not allowed.
-        // The only way to do this is to bind the local to an address on the desired interface,
-        // or bind the socket to the interface itself using SO_BINDTODEVICE.  The latter,
-        // however, requires privileged access.
-        if (addrType == kIPAddressType_IPv4)
+        // Try binding to the interface
+
+        struct ifreq ifr;
+        memset(&ifr, 0, sizeof(ifr));
+
+        res = GetInterfaceName(intf, intfStr, sizeof(intfStr));
```

Comment:
You should eliminate intfStr and have GetInterfaceName copy the interface name directly into ifr.ifr_name.

Reply:
Done. Thanks.